pkg/threescale/connectors/backend/cached_backend.go

```diff
@@ -0,0 +1,261 @@
+package backend
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+
+	backend "github.com/3scale/3scale-go-client/client"
+	"github.com/3scale/3scale-istio-adapter/pkg/threescale/metrics"
+	cmap "github.com/orcaman/concurrent-map"
+)
+
+const (
+	cacheKeySeparator = "_"
+	cacheKey          = "%s" + cacheKeySeparator + "%s"
+)
+
+// Cacheable - defines the required behaviour of a Backend cache
+type Cacheable interface {
+	Get(cacheKey string) (CacheValue, bool)
+	Set(cacheKey string, cv CacheValue)
+}
+
+// Limit captures the current state of the rate limit for a particular time period
+type Limit struct {
+	current int
+	max     int
+	// unix timestamp for end period
+	periodEnds int64
```

Comment:
With regards to the timestamp, that is coming directly from the client, so perhaps we want to make a change there?  Unix time gets returned from stdlib as int64

https://github.com/3scale/3scale-go-client/blob/e366adc214e96578b5e53fd5b0250c5cfaec414c/client/client.go#L242

The other fact here is I don't think we actually use the `periodEnds` field currently so do we even need it?

I'll update the current and max

Reply:
Don't update them just yet because we might end up having them as signed. We'll see.

Re `periodEnds`, yes, probably won't need it in the end. Keep it for now just in case (maybe annotate for reconsideration).